src/workerd/util/mem_utils.c++

```diff
@@ -0,0 +1,37 @@
+#include "mem_utils.h"
+
+#include <kj/debug.h>
+#include <kj/filesystem.h>
+
+#if !_WIN32
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#endif
+
+namespace workerd::util {
+
+#if !_WIN32
```

Comment:
This code is Linux-specific since it depends on procfs.

Reply:
Yes, I know... this bit will need to be fully finished to work on macos/windows before the PR lands.